File: frontend/src/types/wage_analytics.ts
// Wage Analytics types

export interface WageSummary {
  total_wages: number
  total_hours: number
  avg_cost_per_hour: number
}

export interface EmployeeWageStats {
  employee_id: number
  employee_name: string
  position_name: string
  hours_worked: number
  wages_paid: number
  cost_per_hour: number
  previous_month_wages: number | null
  change_percent: number | null
}

export interface WageAnalyticsResponse {
  summary: WageSummary
  previous_month_summary: WageSummary | null
  by_employee: EmployeeWageStats[]
}

export interface HoursCalculationResponse {
  employee_id: number
  hours: number
  calculated_wage: number
}

File: backend/app/schemas/wage_analytics.py
from pydantic import BaseModel
from decimal import Decimal


class WageSummary(BaseModel):
    """Summary of wage statistics for a period."""
    total_wages: Decimal
    total_hours: float
    avg_cost_per_hour: Decimal


class EmployeeWageStats(BaseModel):
    """Wage statistics for a single employee."""
    employee_id: int
    employee_name: str
    position_name: str
    hours_worked: float
    wages_paid: Decimal
    cost_per_hour: Decimal
    previous_month_wages: Decimal | None
    change_percent: float | None


class WageAnalyticsResponse(BaseModel):
    """Complete wage analytics response with summary and per-employee breakdown."""
    summary: WageSummary
    previous_month_summary: WageSummary | None
    by_employee: list[EmployeeWageStats]


class HoursCalculationResponse(BaseModel):
    """Response for hours calculation within a date range."""
    employee_id: int
    hours: float
    calculated_wage: Decimal
